### PR TITLE
Adding vendoring and Travis CI

### PR DESCRIPTION
**Pull request by ryscheng**
This PR adds:
- `govendor` as the tool for vendoring. vendor.json will snapshot the dependencies. golang >=1.5 will know to look in vendor/ automatically, so all we need to do now is clone talek to get it running.
- Added a simple Makefile to encapsulate common tasks
- Added Travis CI with successful builds.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+install: true
+script:
+  - make test
+go:
+  - 1.7
+  - 1.8
+  - master
+notifications:
+  email:
+    - me@raymondcheng.net
```
Comment (willscott) [inline .travis.yml]: happy to have this be a group acct

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+get-tools:
+	go get -u github.com/kardianos/govendor
+	go get -u github.com/alecthomas/gometalinter
+	gometalinter --install
+
+test: unit
+
+lint:
+	gometalinter --vendor --deadline=60s ./...
```
Comment (willscott) [inline Makefile]: add '--tests' to also lint tests

File: README.md [hunk 3]
```diff
@@ -1,10 +1,11 @@
 # Talek
+[![Build Status](https://travis-ci.org/privacylab/talek.svg?branch=master)](https://travis-ci.org/privacylab/talek)
 
 Talek is a privacy-preserving messaging system. User communication is stored on untrusted systems using PIR.
 
 ## Getting Started
 
-A basic client (which is not resistant to traffic analysis!) can be found at
+A basic client (which is not resistant to traffic analysis!) can be found at
```
Comment (willscott) [inline README.md]: space at end of line?

Comment (ryscheng) [inline README.md]: equivalent to <br> in markdown

File: .travis.yml (hunk 1 above)
Comment (ryscheng) [inline .travis.yml]: done

File: Makefile (hunk 2 above)
Comment (ryscheng) [inline Makefile]: done

**Comment by ryscheng** [comment]
Done. Ready to merge?

**Comment by willscott** [comment]
reason not to increase that deadline to like 10 minutes and keep the full suite? since travis already takes a while to schedule jobs, seems plausible

**Comment by ryscheng** [comment]
I've change `make lint` to be the subset that you use, so at least we're all on the same page now. Let's keep the Makefile up to date with whatever new linters we start using.

**Pull request merged**